Test failures due to clang version mismatch

When I run `./scripts/test_translator.py tests` I currently get three test failures. The all have the same root cause: `uint32_t` is not being recognized as a valid type, even though the tests in question include `<stdint.h>`.

I can reproduce the problem with a small example. This C file, `aa.c`:
```c
#include <stdint.h>
int main(void) {
    uint32_t x;
    return 0;
}
```
and this `compile_commands.json` file:
```json
[
  {   
    "arguments": [
        "clang",
        "-I/usr/lib/llvm-19/lib/clang/19/include",
        "aa.c"
    ],
    "directory": "/home/njn/dev/c2rust",
    "file": "aa.c"
  }   
]  
```
and this command:
```
c2rust-transpile compile_commands.json
```
gives this error:
```
Additional Clang arguments: 
aa.c:3:5: error: use of undeclared identifier 'uint32_t'
    3 |     uint32_t x;
      |     ^
1 error generated.
Error while processing /home/njn/dev/c2rust/aa.c.
Transpiling aa.c
```

The problem is with the `-I` line in the `compile_commands.json` file. If I remove it, the problem is fixed. If I change it to `"-I/usr/lib/llvm-18/lib/clang/18/include"`, (i.e. 19 to 18) then problem is also fixed. So it seems like there is some kind of clang version mismatch problem. The `-I` line was added in #892.

I have clang 19.1.1 in my PATH. I also have clang 18 installed. This is on an Ubuntu 24.10 box.

The obvious changes to `c2rust-build-paths/src/lib.rs` fix my `uint32_t` test failures:
```diff
diff --git a/c2rust-build-paths/src/lib.rs b/c2rust-build-paths/src/lib.rs
index e362e501..ed86f916 100644
--- a/c2rust-build-paths/src/lib.rs
+++ b/c2rust-build-paths/src/lib.rs
@@ -91,6 +91,7 @@ pub fn find_llvm_config() -> Option<PathBuf> {
         .or_else(|| {
             // In PATH
             [
+                "llvm-config-19",
                 "llvm-config-18",
                 "llvm-config-17",
                 "llvm-config-16",
@@ -106,6 +107,7 @@ pub fn find_llvm_config() -> Option<PathBuf> {
                 "llvm-config-7.0",
                 "llvm-config",
                 // Homebrew install locations on Intel macOS
+                "/usr/local/opt/llvm@19/bin/llvm-config",
                 "/usr/local/opt/llvm@18/bin/llvm-config",
                 "/usr/local/opt/llvm@17/bin/llvm-config",
                 "/usr/local/opt/llvm@16/bin/llvm-config",
@@ -119,6 +121,7 @@ pub fn find_llvm_config() -> Option<PathBuf> {
                 "/usr/local/opt/llvm@8/bin/llvm-config",
                 "/usr/local/opt/llvm/bin/llvm-config",
                 // Homebrew install locations on Apple Silicon macOS
+                "/opt/homebrew/opt/llvm@19/bin/llvm-config",
                 "/opt/homebrew/opt/llvm@18/bin/llvm-config",
                 "/opt/homebrew/opt/llvm@17/bin/llvm-config",
                 "/opt/homebrew/opt/llvm@16/bin/llvm-config",
```

But now I have different failures relating to atomics, a bunch of errors like this:
```
atomics.c:38:19: error: c2rust: Unknown atomic builtin: 6
   38 |     buffer[i++] = __atomic_fetch_add(&x, 55, __ATOMIC_RELAXED);    buffer[i++] = x;
      |                   ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

--- a/c2rust-ast-exporter/src/AstExporter.cpp
+++ b/c2rust-ast-exporter/src/AstExporter.cpp
@@ -1772,7 +1772,11 @@ class TranslateASTVisitor final
                              case AtomicExpr::AO ## ID:                 \
                                  cbor_encode_string(array, #ID);       \
                                  break;
+#if CLANG_VERSION_MAJOR >= 19
+#include "clang/Basic/Builtins.inc"
+#else
 #include "clang/Basic/Builtins.def"
+#endif
                          default: printError("Unknown atomic builtin: " +
                                              std::to_string(E->getOp()), E);
                          };

--- a/c2rust-build-paths/src/lib.rs
+++ b/c2rust-build-paths/src/lib.rs
@@ -91,6 +91,7 @@ pub fn find_llvm_config() -> Option<PathBuf> {
         .or_else(|| {
             // In PATH
             [
+                "llvm-config-19",
                 "llvm-config-18",
                 "llvm-config-17",
                 "llvm-config-16",
@@ -106,6 +107,7 @@ pub fn find_llvm_config() -> Option<PathBuf> {
                 "llvm-config-7.0",
                 "llvm-config",
                 // Homebrew install locations on Intel macOS
+                "/usr/local/opt/llvm@19/bin/llvm-config",
                 "/usr/local/opt/llvm@18/bin/llvm-config",
                 "/usr/local/opt/llvm@17/bin/llvm-config",
                 "/usr/local/opt/llvm@16/bin/llvm-config",
@@ -119,6 +121,7 @@ pub fn find_llvm_config() -> Option<PathBuf> {
                 "/usr/local/opt/llvm@8/bin/llvm-config",
                 "/usr/local/opt/llvm/bin/llvm-config",
                 // Homebrew install locations on Apple Silicon macOS
+                "/opt/homebrew/opt/llvm@19/bin/llvm-config",
                 "/opt/homebrew/opt/llvm@18/bin/llvm-config",
                 "/opt/homebrew/opt/llvm@17/bin/llvm-config",
                 "/opt/homebrew/opt/llvm@16/bin/llvm-config",

